src/Simulation/Simulators/SparseSimulator/Native/types.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma once
+
+
+#include <vector>
+#include <complex>
+#include <unordered_map>
+#include <bitset>
+#include "quantum_hash_map.hpp"
+#include <shared_mutex>
+#include <mutex>
+
+namespace Microsoft
+{
+	namespace Quantum
+	{
+		namespace SPARSESIMULATOR
+		{
+		using mutex_type = std::mutex;
+		using recursive_mutex_type = std::recursive_mutex;
+		using lock_type = std::lock_guard<mutex_type>;
+		using recursive_lock_type = std::lock_guard<recursive_mutex_type>;
+
+		#ifndef USE_SINGLE_PRECISION
+					using RealType = double;
+		#else
+					using RealType = float;
+		#endif
+
+
+
+		// Logical qubit id is visible to the clients and is immutable during the lifetime of the qubit.
+		using logical_qubit_id = unsigned;
```

Comment:
@sam-jaques, some more info is here:
[Be Careful When Exposing Certain Data Types](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/QIR-RT-API-Design-Guidelines.md#be-careful-when-exposing-certain-data-types)
[Integer Types](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/QIR-RT-API-Design-Guidelines.md#integer-types)

---
Strategically we plan to refactor our QIR Runtime according to [those guidelines](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/QIR-RT-API-Design-Guidelines.md). There is also a chance of refactoring the [full state simulator](https://github.com/microsoft/qsharp-runtime/tree/main/src/Simulation/Native/src/simulator) also. And eventually your SparseSimulator also. So it would be beneficial if you could take into account those guidelines when making fixes to your SparseSimulator.